dashboard/app/controllers/pd/workshop_user_management_controller.rb

```diff
@@ -1,35 +1,68 @@
 class Pd::WorkshopUserManagementController < ApplicationController
   authorize_resource class: :pd_workshop_user_management
 
+  # restrict the PII returned by the controller to the view by selecting only these columns from the model
+  RESTRICTED_USER_ATTRIBUTES_FOR_VIEW = %w(
+    users.id
+    email
+    name
+    user_type
+    current_sign_in_at
+    sign_in_count
+    users.created_at
+  ).freeze
+
   # get /pd/workshop_user_management/facilitator_courses
   def facilitator_courses_form
     search_term = params[:search_term]
     if search_term =~ /^\d+$/
-      @facilitator = User.joins(:permissions).find_by(
-        id: search_term,
-        user_permissions: {permission: UserPermission::FACILITATOR}
-      )
+      @user = restricted_users.find_by(id: search_term)
     elsif search_term
-      @facilitator = User.joins(:permissions).find_by(
-        hashed_email: User.hash_email(search_term),
-        user_permissions: {permission: UserPermission::FACILITATOR}
-      )
+      @user = restricted_users.find_by(hashed_email: restricted_users.hash_email(search_term))
     end
 
-    unless @facilitator || search_term.blank?
-      flash[:notice] = "Facilitator not found"
+    unless @user || search_term.blank?
+      flash[:notice] = "User not found"
     end
   end
 
   # post /pd/workshop_user_management/assign_course
   def assign_course
-    User.find(params[:facilitator_id]).try(:course_as_facilitator=, params[:course])
-    redirect_to action: "facilitator_courses_form", search_term: params[:facilitator_id]
+    @user = restricted_users.find_by(id: params[:user_id])
+    if @user.try(:teacher?)
+      @user.course_as_facilitator = params[:course]
+      @user.permission = UserPermission::FACILITATOR unless @user.facilitator?
+    end
+    redirect_to action: "facilitator_courses_form", search_term: params[:user_id]
   end
 
   # get /pd/workshop_user_management/remove_course
   def remove_course
-    User.find(params[:facilitator_id]).try(:delete_course_as_facilitator, params[:course])
-    redirect_to action: "facilitator_courses_form", search_term: params[:facilitator_id]
+    restricted_users.find(params[:user_id]).try(:delete_course_as_facilitator, params[:course])
+    redirect_to action: "facilitator_courses_form", search_term: params[:user_id]
+  end
+
+  # get /pd/workshop_user_management/update_facilitator_permission
+  def update_facilitator_permission
+    @user = restricted_users.find_by(id: params[:user_id])
+    if @user.try(:teacher?)
```

Comment:
Why `try` here? If @user is nil, find_by will have already thrown an exception.

I think this method is unnecessary - we shouldn't be designating anyone as a facilitator without assigning a certain course to them. And if we want to de-facilitator-them, then remove all the course facilitators (or have a button in the UI that destroys all of them).